content/6_Advanced/Eulers_Formula.mdx

```diff
@@ -1,104 +1,189 @@
 ---
 id: eulers-formula
 title: "Euler's Formula"
-author: Benjamin Qi
+author: Benjamin Qi, Alexis Tao
 description: A formula for finding the number of faces in a planar graph.
 prerequisites:
   - dsu
   - 2DRQ
 frequency: 1
 ---
 
-## Introduction
+Euler's Formula, a fundamental gem in mathematics, uncovers a deep connection between the vertices, edges, and faces of 3D shapes. It states that for any polyhedron, the number of vertices minus the number of edges plus the number of faces always equals 2. This seemingly simple equation is used to explore and understand complex shapes with precision and insight. Euler's formula for planar graphs is given by V−E+F=2, where V is the number of vertices, E is the number of edges, and F is the number of faces (including the exterior).
 
-<IncompleteSection />
+## Island Archipelago
 
-## Example 1
+<FocusProblem problem="island" />
 
-<Problems problems="e1" />
+### Explanation
 
-<IncompleteSection />
+In this problem, Euler's Formula is indirectly relevant through the concept of islands in a grid-based archipelago. Each island can be seen as a region of connected land cells. Euler's Formula, $V−E+F=1$, relates the vertices (land cells), edges (shared boundaries between land cells), and faces (islands) of a planar graph. Initially, with only water, there are no islands (F = 0). As land cells appear and disappear due to changing water levels (modeled by queries), islands form and potentially enclose freshwater lakes. The challenge is to maintain a dynamic count of islands and determine if each island contains a freshwater lake. This involves efficiently managing and updating connected components of land cells and checking for closed loops (cycles) of land cells that surround at least one water cell to identify lakes. The queries ('!' to change water levels and '?' to request current island statistics) dynamically alter the state of the archipelago and prompt recalculations of island counts and freshwater lake checks.
 
-## Example 2
+## Implementation
 
-<Problems problems="e2" />
+**Steps**
+1. DisjointSets: Implement a union-find structure to manage connected components (islands) efficiently using path compression and union by size.
+2. Grid Representation: Use a 2D boolean vector to represent the archipelago, where true indicates land and false indicates water. Each cell can be accessed via a 1D index for the DSU.
+3. Handling Updates (! r c): If converting water to land, increase the island count and unite with neighboring land cells. Check for lakes. If converting land to water, decrease the island count and mark the component as having no lake.
+4. Handling Queries (?): Returns the total number of islands and the number of islands without lakes.
```

Comment:
![image](https://github.com/user-attachments/assets/b108cd36-661f-45f5-8d15-b4b2486257cd)

Reply:
Would you like me to articulate the steps to the solution differently. I'm definitely not advance enough in USACO to properly give steps for how I would solve this solution.